modules/actions/workflows.go

```diff
@@ -69,6 +71,42 @@ func ListWorkflows(commit *git.Commit) (git.Entries, error) {
 	return ret, nil
 }
 
+// GetRunsWorkflowFileLink return url for the source of the workflow file for an ActionRun
+func GetRunsWorkflowFileLink(run *actions_model.ActionRun, gitRepo *git.Repository) string {
+	if run.Repo == nil || run.CommitSHA == "" {
+		return ""
+	}
+
+	commit, err := gitRepo.GetCommit(run.CommitSHA)
+	if err != nil {
+		return ""
+	}
+
+	entries, err := ListWorkflows(commit)
```

Comment:
OK, I think I could understand the problem now.

The proper approach should be like this: add a "WorkflowFilePath" column into the "Actions" model and just use it.

It shouldn't call `ListWorkflows` for viewing action workflows, otherwise there will be a performance problem.